conversion/indico_conversion/conversion.py

```diff
@@ -98,20 +102,80 @@ def _process(self):
-        pdf_attachment.file = AttachmentFile(user=attachment.file.user, filename=f'{name}.pdf',
-                                             content_type='application/pdf')
-        pdf_attachment.file.save(data)
-        db.session.add(pdf_attachment)
-        db.session.flush()
-        pdf_state_cache.set(str(attachment.id), 'finished', timeout=timedelta(minutes=15))
-        ConversionPlugin.logger.info('Added PDF attachment %s for %s', pdf_attachment, attachment)
-        signals.attachments.attachment_created.send(pdf_attachment, user=None)
+        pdf = request.files['content'].stream.read()
+        save_pdf(attachment, pdf)
+        return jsonify(success=True)
+
+
+@celery.task(bind=True, max_retries=3)
+def submit_attachment_cloudconvert(task, attachment):
+    """Sends an attachment's file to the CloudConvert conversion service"""
+    from indico_conversion.plugin import ConversionPlugin
+    if ConversionPlugin.settings.get('maintenance'):
+        task.retry(countdown=900)
+
+    cloudconvert.configure(api_key=ConversionPlugin.settings.get('cloudconvert_api_key'), sandbox=True)
+
+    job = cloudconvert.Job.create(payload={
+        "tag": str(attachment.id),
+        "tasks": {
+            'import-my-file': {
+                'operation': 'import/upload',
+            },
+            'convert-my-file': {
+                'operation': 'convert',
+                'input': 'import-my-file',
+                'output_format': 'pdf',
+            },
+            'export-my-file': {
+                'operation': 'export/url',
+                'input': 'convert-my-file'
+            }
+        },
+        "webhook_url": 'https://aa02-2001-1458-202-3-00-100-35.eu.ngrok.io/conversion/cloudconvert/finished'
+        # "webhook_url": url_for_plugin('conversion.cloudconvert-callback', _external=True)
+    })
+
+    upload_task_id = job['tasks'][0]['id']
+    task = cloudconvert.Task.find(id=upload_task_id)
+    form = task.get('result').get('form')
+    with attachment.file.open() as fd:
+        try:
+            response = requests.request(method='POST', url=form.get('url'),
+                                        files={'file': fd}, data=form.get('parameters'))
+            response.raise_for_status()
+            if response.status_code != 201:
+                raise requests.RequestException(f'Unexpected response status from server: {response.status_code}')
+        except requests.RequestException as exc:
+            retry_task(task, attachment, exc)
+        else:
+            ConversionPlugin.logger.info('Submitted %r', attachment)
+
+
+class RHCloudConvertFinished(RH):
+    """Callback to attach a converted file"""
+
+    CSRF_ENABLED = False
+
+    def _process(self):
+        from indico_conversion.plugin import ConversionPlugin
+
+        event = request.json['event']
+        if event == 'job.failed':
+            ConversionPlugin.logger.error(f'Cloudconvert conversion job failed: {event}')
```

Comment:
`.error('whatever: %s', event)` so we get proper grouping on sentry.

Even though in this case passing `event` may not be that useful in the first place because it's always `job.failed`